Runtime/Helpers/QuickOutline/Scripts/Outline.cs

```diff
@@ -0,0 +1,350 @@
+                var smoothNormals = (index >= 0) ? bakeValues[index].data : SmoothNormals(meshFilter.sharedMesh);
+
+                // Store smooth normals in UV3
+                meshFilter.sharedMesh.SetUVs(3, smoothNormals);
+
+                // Combine submeshes
+                var renderer = meshFilter.GetComponent<Renderer>();
+
+                if (renderer != null)
+                {
+                    CombineSubmeshes(meshFilter.sharedMesh, renderer.sharedMaterials);
+                }
+            }
+
+            // Clear UV3 on skinned mesh renderers
+            foreach (var skinnedMeshRenderer in GetComponentsInChildren<SkinnedMeshRenderer>())
+            {
+
+                // Skip if UV3 has already been reset
+                if (!registeredMeshes.Add(skinnedMeshRenderer.sharedMesh))
+                {
+                    continue;
+                }
+
+                // Clear UV3
+                skinnedMeshRenderer.sharedMesh.uv4 = new Vector2[skinnedMeshRenderer.sharedMesh.vertexCount];
+
+                // Combine submeshes
+                CombineSubmeshes(skinnedMeshRenderer.sharedMesh, skinnedMeshRenderer.sharedMaterials);
+            }
+        }
+
+        List<Vector3> SmoothNormals(Mesh mesh)
+        {
+
+            // Group vertices by location
+            var groups = mesh.vertices.Select((vertex, index) => new KeyValuePair<Vector3, int>(vertex, index)).GroupBy(pair => pair.Key);
+
+            // Copy normals to a new list
+            var smoothNormals = new List<Vector3>(mesh.normals);
+
+            // Average normals for grouped vertices
+            foreach (var group in groups)
+            {
+
+                // Skip single vertices
+                if (group.Count() == 1)
+                {
+                    continue;
+                }
+
+                // Calculate the average normal
+                var smoothNormal = Vector3.zero;
+
+                foreach (var pair in group)
+                {
+                    smoothNormal += smoothNormals[pair.Value];
+                }
+
+                smoothNormal.Normalize();
+
+                // Assign smooth normal to each vertex
+                foreach (var pair in group)
+                {
+                    smoothNormals[pair.Value] = smoothNormal;
+                }
+            }
+
+            return smoothNormals;
+        }
+
+        void CombineSubmeshes(Mesh mesh, Material[] materials)
+        {
+
+            // Skip meshes with a single submesh
+            if (mesh.subMeshCount == 1)
+            {
+                return;
+            }
+
+            // Skip if submesh count exceeds material count
+            if (mesh.subMeshCount > materials.Length)
+            {
+                return;
+            }
+
+            // Append combined submesh
+            mesh.subMeshCount++;
+            mesh.SetTriangles(mesh.triangles, mesh.subMeshCount - 1);
+        }
+
+        void UpdateMaterialProperties()
+        {
+
```

Comment:
Maybe implement this variables
```suggestion
    float zTestAlways = (float)UnityEngine.Rendering.CompareFunction.Always;
    float zTestLessEqual = (float)UnityEngine.Rendering.CompareFunction.LessEqual;
    float zTestGreater = (float)UnityEngine.Rendering.CompareFunction.Greater;
```